### ISSUE TITLE
replace jiter dependency with build-in pydantic function

### ISSUE THREAD

**Issue by blshkv** [openai/openai-python#1616]
### Confirm this is an issue with the Python library and not an underlying OpenAI API

- [X] This is an issue with the Python library

### Describe the bug

https://pypi.org/project/jiter/
```
This is a standalone version of the JSON parser used in pydantic-core
The recommendation is to only use this package directly if you do not use pydantic.
```

The reason of this request is some OS (Gentoo in my case) do provide both ```pydantic``` and ```pydantic-core``` libraries, but ```jiter```

openai-python uses ```pydantic``` already, so it should be possible to re-use build-in function (perhaps, via ```pydantic-core```)

### To Reproduce

1. inspect this line: https://github.com/openai/openai-python/blob/main/pyproject.toml#L19

### Code snippets

```Python
https://github.com/openai/openai-python/blob/main/src/openai/lib/streaming/chat/_completions.py
 jiter import from_json
```

### OS

any

### Python version

any

### Library version

1.40.0

**Comment by RobertCraigie** [comment]
We need to use the standalone package because we support both Pydantic v1 and Pydantic v2. Unfortunately pydantic v1 does not provide the same JSON parsing functionality.

> The reason of this request is some OS (Gentoo in my case) do provide both pydantic and pydantic-core libraries, but jiter

What do you mean? How are you managing dependencies?

**Comment by blshkv** [comment]
You should probably consider stopping supporting pedantic 1 at some point, for simplicity reason.
jiter is used just once in the code.

https://docs.pydantic.dev/latest/version-policy/
```
Active development of V1 has already stopped, however critical bug fixes and security vulnerabilities will be fixed in V1 for one year after the release of V2 (June 30, 2024).
```

FYI, Gentoo provides pedantic 2 only (https://packages.gentoo.org/packages/dev-python/pydantic )
and pedantic-core (version 2 as well)

I had to create my own jiter ebuild, but it is complicated, since it is a mixture of rust and python. And it's basically similar with pydantic-core.

**Comment by RobertCraigie** [comment]
Unfortunately we have no plans to drop support for Pydantic v1 anytime soon as it's still getting many millions of downloads a [day](https://www.pepy.tech/projects/pydantic?versions=2.*&versions=1.*).

Please ask Gentoo to provide a packaged version of `jiter` if this is important to you.

**Comment by dinopanagos** [comment]
do we have an arn for jiter we can reference for python 312?
i cant stop getting no module named jiter.jiter
even with pydantic and openai in my req txt

**Comment by blshkv** [comment]
> do we have an arn for jiter we can reference for python 312? i cant stop getting no module named jiter.jiter even with pydantic and openai in my req txt

You need either outdated pydantic 1 or install jiter (https://pypi.org/project/jiter/)

@RobertCraigie please lock this issue, so people won't post unrelated comments.

**Comment by dinopanagos** [comment]
id say this is pretty related... what are you on
if install pydantic/openai it automatically installs jiter and throws the error.

theres 50+ versions of pydantic 1 how could i possible know which outdated would work lol

**Comment by phazei** [comment]
having to install the entire rust toolchain just to install jiter for one call seems excessive and a drastically inefficient choice of dependency.  Couldn't you just copy over the method or use anything native?  parsing a line of json must be possible without rust...

**Comment by icarito** [comment]
Yeah I'm trying to compile this in Haiku OS and I'm running into the same problem, Rust is having issues compiling jiter.

**Comment by barracuda156** [comment]
Also: https://github.com/openai/openai-python/issues/2102
